### ISSUE TITLE
Bikeshed Scout Options

### ISSUE THREAD

**Issue by o0Ignition0o** [o0Ignition0o/cargo-scout#59]
As explained [here](https://github.com/o0Ignition0o/cargo-scout/pull/57#discussion_r359515227) let's bikeshed ScoutOptions and figure out a proper way to link Linter, Config and Options in a Cargo Scout so we have the best traits and impls !

**Comment by ebroto** [comment]
So just to make clear what I was proposing in the PR, I think one possibility is to instantiate `linter::clippy::Clippy` and `project::cargo::Config` in `main.rs`, and have `run` defined in `lib.rs` along this lines:
```
fn run<L: Linter, C: Config>(linter: L, config: C) -> Result<Vec<Lint>, Error> {
    // ...
}
``` 
Then `run` could `set_project_config` and `set_linter` on an instance of `Scout`, regardless of the types of the linter and project config.

**Comment by o0Ignition0o** [comment]
That makes total sense, let's do this first, and then bikeshed our way through an option struct that makes sense !

**Comment by o0Ignition0o** [comment]
This issue has been fixed by #61 a while ago.
